internal/controllers/sopssecret_controller.go

```diff
@@ -35,10 +35,11 @@ import (
 // SopsSecretReconciler reconciles a SopsSecret object.
 type SopsSecretReconciler struct {
 	client.Client
-	Metrics  *metrics.Recorder
-	Log      logr.Logger
-	Recorder record.EventRecorder
-	Scheme   *runtime.Scheme
+	Metrics      *metrics.Recorder
+	Log          logr.Logger
+	Recorder     record.EventRecorder
+	Scheme       *runtime.Scheme
+	EnableStatus bool
```

Comment:
Where are you passing this from the cmd? Currently it's always gonna be false.

I recommend creating a dedicated config struct:

```go
type SopsSecretReconcilerConfig struct {
   EnableStatus bool
}
```

Then referencing the struct in the `SopsSecretReconciler struct`:

```go
config SopsSecretReconcilerConfig
```

Then pass the config on reconciler setup:

```go
// SetupWithManager sets up the controller with the Manager.
func (r *SopsSecretReconciler) SetupWithManager(mgr ctrl.Manager, cfg SopsSecretReconcilerConfig) error {
        r.config = cfg

	return ctrl.NewControllerManagedBy(mgr).
		For(&sopsv1alpha1.SopsSecret{}).
		Watches(&corev1.Secret{},
```

Now you need to pass the config from the cmd package